api_tests/caching/test_caching.py

```diff
@@ -0,0 +1,236 @@
+
+                varnish_resp = requests.get(varnish_url, timeout=120)
+
+                python_authed_resp = requests.get(python_url, auth=self.authorization, timeout=120)
+
+                varnish_authed_resp = requests.get(varnish_url, auth=self.authorization, timeout=120)
+
+
+                try:
+                    python_data[key]['_'.join(embed_values)] = python_resp.json()
+                    self.validate_keys(python_resp.json(), original_embed_values)
+                except AssertionError as ex:
+                    print 'Validation failed python_url unauthed: {} - {}\n\n\n'.format(python_url, ex.message)
+                except Exception as ex:
+                    python_data[key]['_'.join(embed_values)] = dict(error=ex.message)
+                    print 'Failed to get python_url unauthed: {} - {}\n\n\n'.format(python_url, ex.message)
+
+                try:
+                    python_authed_data[key]['_'.join(embed_values)] = python_authed_resp.json()
+                    self.validate_keys(python_authed_resp.json(), original_embed_values)
+                except AssertionError as ex:
+                    print 'Validation failed python_url authed: {} - {}\n\n\n'.format(python_url, ex.message)
+                except Exception as ex:
+                    python_authed_data[key]['_'.join(embed_values)] = dict(error=ex.message)
+                    print 'Failed to get python_url authed: {} - {}\n\n\n'.format(python_url, ex.message)
+
+                try:
+                    varnish_data[key]['_'.join(embed_values)] = varnish_resp.json()
+                    self.validate_keys(varnish_resp.json(), original_embed_values)
+                except AssertionError as ex:
+                    print 'Validation failed varnish_url unauthed: {} - {}\n\n\n'.format(varnish_url, ex.message)
+                except Exception as ex:
+                    varnish_data[key]['_'.join(embed_values)] = dict(error=ex.message)
+                    print 'Failed to get and validate varnish_url unauthed: {} - {}\n\n\n'.format(varnish_url,
+                                                                                                  ex.message)
+
+                try:
+                    varnish_authed_data[key]['_'.join(embed_values)] = varnish_authed_resp.json()
+                    self.validate_keys(varnish_authed_resp.json(), original_embed_values)
+                except AssertionError as ex:
+                    print 'Validation failed varnish_url authed: {} - {}\n\n\n'.format(varnish_url, ex.message)
+                except Exception as ex:
+                    varnish_authed_data[key]['_'.join(embed_values)] = dict(error=ex.message)
+                    print 'Failed to get and validate varnish_url authed: {} - {}\n\n\n'.format(varnish_url, ex.message)
+
+                embed_values.pop()
+
+        # Uncomment these lines to write out files containing responses for comparison
```

Comment:
rgr, I'll remove it. I didn't put datadiff in the requirements because that was it's only usage so uncommenting it would fail anyway.